src/SprykerEco/Zed/PunchoutCatalogs/Communication/Table/PunchoutCatalogsConnectionsTable.php

```diff
@@ -0,0 +1,196 @@
+            return [];
+        }
+
+        return $this->mapPunchoutCatalogConnections($punchoutCatalogConnectionCollection);
+    }
+
+    /**
+     * @module Company
+     *
+     * @param \Orm\Zed\PunchoutCatalog\Persistence\PgwPunchoutCatalogConnectionQuery
+     *
+     * @return \Orm\Zed\PunchoutCatalog\Persistence\PgwPunchoutCatalogConnectionQuery
+     */
+    protected function prepareQuery(PgwPunchoutCatalogConnectionQuery $connectionPropelQuery): PgwPunchoutCatalogConnectionQuery
+    {
+        return $connectionPropelQuery->joinWithCompanyBusinessUnit()
+            ->useCompanyBusinessUnitQuery()
+            ->joinCompany()
+                ->withColumn(SpyCompanyTableMap::COL_NAME, static::COL_COMPANY)
+            ->endUse();
+    }
+
+    /**
+     * @param \Propel\Runtime\Collection\ObjectCollection|\Orm\Zed\PunchoutCatalog\Persistence\PgwPunchoutCatalogConnection[] $punchoutCatalogConnectionCollection
+     *
+     * @return array
+     */
+    protected function mapPunchoutCatalogConnections(ObjectCollection $punchoutCatalogConnectionCollection): array
+    {
+        $punchoutCatalogConnectionRows = [];
+
+        foreach ($punchoutCatalogConnectionCollection as $punchoutCatalogConnection) {
+            $punchoutCatalogConnectionRows[] = $this->mapPunchoutCatalogConnection($punchoutCatalogConnection);
+        }
+
+        return $punchoutCatalogConnectionRows;
+    }
+
+    /**
+     * @param \Orm\Zed\PunchoutCatalog\Persistence\PgwPunchoutCatalogConnection $punchoutCatalogConnection
+     *
+     * @return array
+     */
+    protected function mapPunchoutCatalogConnection(PgwPunchoutCatalogConnection $punchoutCatalogConnection): array
+    {
+        $punchoutCatalogConnectionRow = $punchoutCatalogConnection->toArray();
+
+        $punchoutCatalogConnectionRow[static::ACTIONS] = $this->buildLinks($punchoutCatalogConnection);
+        $punchoutCatalogConnectionRow[static::COL_CREATED_AT] = $this->utilDateTimeService->formatDateTime($punchoutCatalogConnection->getCreatedAt());
+        $punchoutCatalogConnectionRow[static::COL_STATUS] = $this->getStatusLabel($punchoutCatalogConnection);
+        $punchoutCatalogConnectionRow[static::COL_FORMAT] = $this->generateLabel($punchoutCatalogConnection->getFormat(), 'label-secondary');
+        $punchoutCatalogConnectionRow[static::COL_COMPANY] = $punchoutCatalogConnection->getCompanyBusinessUnit()
+            ->getCompany()
+            ->getName();
+
+        return $punchoutCatalogConnectionRow;
+    }
+
+    /**
+     * @param \Orm\Zed\PunchoutCatalog\Persistence\PgwPunchoutCatalogConnection $punchoutCatalogConnection
+     *
+     * @return string
+     */
+    protected function buildLinks(PgwPunchoutCatalogConnection $punchoutCatalogConnection): string
+    {
+        $buttons = [];
+        $buttons[] = $this->generateEditButton('NOT_IMPLEMENTED', 'Activate');
+        $buttons[] = $this->generateEditButton('NOT_IMPLEMENTED', 'Test Cart');
```

Comment:
@asmarovydlo I might be wrong, but I guess Karoly said we won't need this button?

Reply:
Sorry, missed this one, @gerner-spryker could you please confirm?

Reply:
Correct, we are not going to have it in the first version. It only comes later.